mantle/platform/qemu.go

```diff
@@ -587,14 +594,24 @@ func (builder *QemuBuilder) setupAdditionalNetworking() error {
 		macSuffix := fmt.Sprintf("%02x", macCounter)
 
 		netdev := fmt.Sprintf("user,id=eth%s,dhcpstart=10.0.2.%s", idSuffix, netSuffix)
-		device := virtio("net", fmt.Sprintf("netdev=eth%s,mac=52:55:00:d1:56:%s", idSuffix, macSuffix))
+		device := virtio(builder.architecture, "net", fmt.Sprintf("netdev=eth%s,mac=52:55:00:d1:56:%s", idSuffix, macSuffix))
 		builder.Append("-netdev", netdev, "-device", device)
 		macCounter++
 	}
 
 	return nil
 }
 
+// SetArchitecture enables qemu full emulation for the target architecture.
+func (builder *QemuBuilder) SetArchitecture(arch string) error {
+	switch arch {
+	case "x86_64", "aarch64", "s390x", "ppc64le":
+		return fmt.Errorf("architecture %s not supported by coreos-assembler qemu", arch)
+	}
+	builder.architecture = arch
```

Comment:
This looks off. Shouldn't it be the opposite?

Reply:
Yes, good catch!  Looks like I forgot to `git commit --amend` here.